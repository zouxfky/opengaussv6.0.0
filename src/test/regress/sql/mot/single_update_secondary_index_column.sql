--T1
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
update test_update_column set y=3 where x=1;
select * from test_update_column;
update test_update_column set z=4 where x=1;
select * from test_update_column;

--T2
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
begin;
update test_update_column set y=3 where x=1;
select * from test_update_column;
update test_update_column set z=4 where x=1;
select * from test_update_column;
commit;

--T3
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
begin;
insert into test_update_column values (1,2,3,100);
update test_update_column set y=3 where x=1;
select * from test_update_column;
update test_update_column set z=4 where x=1;
select * from test_update_column;
commit;

--T4
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
begin;
delete from test_update_column;
insert into test_update_column values (1,2,3,100);
update test_update_column set y=3 where x=1;
select * from test_update_column;
update test_update_column set z=4 where x=1;
select * from test_update_column;
commit;
select * from test_update_column;

--T5
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,4,100);
begin;
update test_update_column set y=3 where x=1;
select * from test_update_column;
update test_update_column set z=4 where x=1;
select * from test_update_column;
commit;
select * from test_update_column;

--T6
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
begin;
update test_update_column set y=3 where x=1;
select * from test_update_column;
update test_update_column set z=4 where x=1;
select * from test_update_column;
insert into test_update_column values (3,3,4,100);
commit;
select * from test_update_column;

--T7
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
begin;
select * from test_update_column;
update test_update_column set y=3 where x=1;
select * from test_update_column;
update test_update_column set z=5 where x=1;
select * from test_update_column;
insert into test_update_column values (3,3,4,100);
commit;
select * from test_update_column;

--T8
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
begin;
select * from test_update_column;
update test_update_column set y=3 where x=1;
select * from test_update_column;
update test_update_column set z=5 where x=1;
select * from test_update_column;
insert into test_update_column values (3,3,4,100);
select * from test_update_column;
update test_update_column set y=100 where y=3;
commit;
select * from test_update_column;

--T9
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,4,100);
begin;
select * from test_update_column;
update test_update_column set z=5;
commit;
select * from test_update_column;

--T10
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
update test_update_column set y=3 where x=1;
alter foreign table test_update_column drop column y;
select * from test_update_column;
update test_update_column set z=4 where x=1;
select * from test_update_column;

--T11
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
update test_update_column set y=3 where x=1;
select * from test_update_column;
update test_update_column set z=4 where x=1;
select * from test_update_column;
alter foreign table test_update_column drop column z;
select * from test_update_column;

--T12
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
begin;
update test_update_column set y=3 where x=1;
select * from test_update_column;
update test_update_column set z=4 where x=1;
select * from test_update_column;
alter foreign table test_update_column drop column z;
commit;
select * from test_update_column;

--T13
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
begin;
update test_update_column set y=3 where x=1;
select * from test_update_column;
alter foreign table test_update_column drop column z;
update test_update_column set z=4 where x=1;
select * from test_update_column;
commit;
select * from test_update_column;

--T14
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
begin;
delete from test_update_column;
insert into test_update_column values (1,2,3,100);
update test_update_column set y=3 where x=1;
select * from test_update_column;
update test_update_column set z=4 where x=1;
select * from test_update_column;
alter foreign table test_update_column drop column z;
commit;
select * from test_update_column;

--T15
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
begin;
select * from test_update_column;
update test_update_column set y=3 where x=1;
select * from test_update_column;
update test_update_column set z=5 where x=1;
select * from test_update_column;
insert into test_update_column values (3,3,4,100);
select * from test_update_column;
update test_update_column set y=100 where y=3;
select * from test_update_column;
alter foreign table test_update_column drop column y;
commit;
select * from test_update_column;

--T16
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,4,100);
begin;
delete from test_update_column;
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,5,100);
update test_update_column set z=z+1;
commit;
select * from test_update_column;

--T17
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,4,100);
begin;
delete from test_update_column;
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,5,100);
update test_update_column set z=z+1;
alter foreign table test_update_column drop column y;
commit;
select * from test_update_column;


--T18
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,4,100);
begin;
delete from test_update_column;
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,5,100);
update test_update_column set z=z+1;
delete from test_update_column;
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,5,100);
update test_update_column set z=z+1;
commit;
select * from test_update_column;

--T19
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,4,100);
begin;
delete from test_update_column;
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,5,100);
update test_update_column set z=z+1;
delete from test_update_column;
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,5,100);
update test_update_column set z=z+1;
alter foreign table test_update_column drop column y;
commit;
select * from test_update_column;

drop foreign table test_update_column;
