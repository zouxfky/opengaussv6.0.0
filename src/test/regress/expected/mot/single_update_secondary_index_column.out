--T1
drop foreign table test_update_column;
ERROR:  foreign table "test_update_column" does not exist
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
NOTICE:  CREATE FOREIGN TABLE / PRIMARY KEY will create constraint "test_update_column_pkey" for foreign table "test_update_column"
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
update test_update_column set y=3 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 3 |  100
(1 row)

update test_update_column set z=4 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 4 |  100
(1 row)

--T2
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
NOTICE:  CREATE FOREIGN TABLE / PRIMARY KEY will create constraint "test_update_column_pkey" for foreign table "test_update_column"
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
begin;
update test_update_column set y=3 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 3 |  100
(1 row)

update test_update_column set z=4 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 4 |  100
(1 row)

commit;
--T3
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
NOTICE:  CREATE FOREIGN TABLE / PRIMARY KEY will create constraint "test_update_column_pkey" for foreign table "test_update_column"
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
begin;
insert into test_update_column values (1,2,3,100);
update test_update_column set y=3 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 3 |  100
(1 row)

update test_update_column set z=4 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 4 |  100
(1 row)

commit;
--T4
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
NOTICE:  CREATE FOREIGN TABLE / PRIMARY KEY will create constraint "test_update_column_pkey" for foreign table "test_update_column"
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
begin;
delete from test_update_column;
insert into test_update_column values (1,2,3,100);
update test_update_column set y=3 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 3 |  100
(1 row)

update test_update_column set z=4 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 4 |  100
(1 row)

commit;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 4 |  100
(1 row)

--T5
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
NOTICE:  CREATE FOREIGN TABLE / PRIMARY KEY will create constraint "test_update_column_pkey" for foreign table "test_update_column"
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,4,100);
begin;
update test_update_column set y=3 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 3 |  100
 3 | 3 | 4 |  100
(2 rows)

update test_update_column set z=4 where x=1;
ERROR:  duplicate key value violates unique constraint "idx2"
DETAIL:  Key (z)=(4) already exists.
select * from test_update_column;
ERROR:  current transaction is aborted, commands ignored until end of transaction block, firstChar[Q]
commit;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 2 | 3 |  100
 3 | 3 | 4 |  100
(2 rows)

--T6
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
NOTICE:  CREATE FOREIGN TABLE / PRIMARY KEY will create constraint "test_update_column_pkey" for foreign table "test_update_column"
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
begin;
update test_update_column set y=3 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 3 |  100
(1 row)

update test_update_column set z=4 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 4 |  100
(1 row)

insert into test_update_column values (3,3,4,100);
ERROR:  duplicate key value violates unique constraint "idx2"
DETAIL:  Key (z)=(4) already exists.
commit;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 2 | 3 |  100
(1 row)

--T7
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
NOTICE:  CREATE FOREIGN TABLE / PRIMARY KEY will create constraint "test_update_column_pkey" for foreign table "test_update_column"
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
begin;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 2 | 3 |  100
(1 row)

update test_update_column set y=3 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 3 |  100
(1 row)

update test_update_column set z=5 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 5 |  100
(1 row)

insert into test_update_column values (3,3,4,100);
commit;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 5 |  100
 3 | 3 | 4 |  100
(2 rows)

--T8
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
NOTICE:  CREATE FOREIGN TABLE / PRIMARY KEY will create constraint "test_update_column_pkey" for foreign table "test_update_column"
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
begin;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 2 | 3 |  100
(1 row)

update test_update_column set y=3 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 3 |  100
(1 row)

update test_update_column set z=5 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 5 |  100
(1 row)

insert into test_update_column values (3,3,4,100);
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 5 |  100
 3 | 3 | 4 |  100
(2 rows)

update test_update_column set y=100 where y=3;
commit;
select * from test_update_column;
 x |  y  | z | data 
---+-----+---+------
 1 | 100 | 5 |  100
 3 | 100 | 4 |  100
(2 rows)

--T9
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
NOTICE:  CREATE FOREIGN TABLE / PRIMARY KEY will create constraint "test_update_column_pkey" for foreign table "test_update_column"
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,4,100);
begin;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 2 | 3 |  100
 3 | 3 | 4 |  100
(2 rows)

update test_update_column set z=5;
ERROR:  duplicate key value violates unique constraint "idx2"
DETAIL:  Key (z)=(5) already exists.
commit;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 2 | 3 |  100
 3 | 3 | 4 |  100
(2 rows)

--T10
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
NOTICE:  CREATE FOREIGN TABLE / PRIMARY KEY will create constraint "test_update_column_pkey" for foreign table "test_update_column"
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
update test_update_column set y=3 where x=1;
alter foreign table test_update_column drop column y;
select * from test_update_column;
 x | z | data 
---+---+------
 1 | 3 |  100
(1 row)

update test_update_column set z=4 where x=1;
select * from test_update_column;
 x | z | data 
---+---+------
 1 | 4 |  100
(1 row)

--T11
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
NOTICE:  CREATE FOREIGN TABLE / PRIMARY KEY will create constraint "test_update_column_pkey" for foreign table "test_update_column"
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
update test_update_column set y=3 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 3 |  100
(1 row)

update test_update_column set z=4 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 4 |  100
(1 row)

alter foreign table test_update_column drop column z;
select * from test_update_column;
 x | y | data 
---+---+------
 1 | 3 |  100
(1 row)

--T12
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
NOTICE:  CREATE FOREIGN TABLE / PRIMARY KEY will create constraint "test_update_column_pkey" for foreign table "test_update_column"
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
begin;
update test_update_column set y=3 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 3 |  100
(1 row)

update test_update_column set z=4 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 4 |  100
(1 row)

alter foreign table test_update_column drop column z;
commit;
select * from test_update_column;
 x | y | data 
---+---+------
 1 | 3 |  100
(1 row)

--T13
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
NOTICE:  CREATE FOREIGN TABLE / PRIMARY KEY will create constraint "test_update_column_pkey" for foreign table "test_update_column"
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
begin;
update test_update_column set y=3 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 3 |  100
(1 row)

alter foreign table test_update_column drop column z;
update test_update_column set z=4 where x=1;
ERROR:  column "z" of relation "test_update_column" does not exist
LINE 1: update test_update_column set z=4 where x=1;
                                      ^
select * from test_update_column;
ERROR:  current transaction is aborted, commands ignored until end of transaction block, firstChar[Q]
commit;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 2 | 3 |  100
(1 row)

--T14
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
NOTICE:  CREATE FOREIGN TABLE / PRIMARY KEY will create constraint "test_update_column_pkey" for foreign table "test_update_column"
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
begin;
delete from test_update_column;
insert into test_update_column values (1,2,3,100);
update test_update_column set y=3 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 3 |  100
(1 row)

update test_update_column set z=4 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 4 |  100
(1 row)

alter foreign table test_update_column drop column z;
commit;
select * from test_update_column;
 x | y | data 
---+---+------
 1 | 3 |  100
(1 row)

--T15
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
NOTICE:  CREATE FOREIGN TABLE / PRIMARY KEY will create constraint "test_update_column_pkey" for foreign table "test_update_column"
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
begin;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 2 | 3 |  100
(1 row)

update test_update_column set y=3 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 3 |  100
(1 row)

update test_update_column set z=5 where x=1;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 5 |  100
(1 row)

insert into test_update_column values (3,3,4,100);
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 3 | 5 |  100
 3 | 3 | 4 |  100
(2 rows)

update test_update_column set y=100 where y=3;
select * from test_update_column;
 x |  y  | z | data 
---+-----+---+------
 1 | 100 | 5 |  100
 3 | 100 | 4 |  100
(2 rows)

alter foreign table test_update_column drop column y;
commit;
select * from test_update_column;
 x | z | data 
---+---+------
 1 | 5 |  100
 3 | 4 |  100
(2 rows)

--T16
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
NOTICE:  CREATE FOREIGN TABLE / PRIMARY KEY will create constraint "test_update_column_pkey" for foreign table "test_update_column"
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,4,100);
begin;
delete from test_update_column;
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,5,100);
update test_update_column set z=z+1;
commit;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 2 | 4 |  100
 3 | 3 | 6 |  100
(2 rows)

--T17
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
NOTICE:  CREATE FOREIGN TABLE / PRIMARY KEY will create constraint "test_update_column_pkey" for foreign table "test_update_column"
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,4,100);
begin;
delete from test_update_column;
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,5,100);
update test_update_column set z=z+1;
alter foreign table test_update_column drop column y;
commit;
select * from test_update_column;
 x | z | data 
---+---+------
 1 | 4 |  100
 3 | 6 |  100
(2 rows)

--T18
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
NOTICE:  CREATE FOREIGN TABLE / PRIMARY KEY will create constraint "test_update_column_pkey" for foreign table "test_update_column"
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,4,100);
begin;
delete from test_update_column;
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,5,100);
update test_update_column set z=z+1;
delete from test_update_column;
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,5,100);
update test_update_column set z=z+1;
commit;
select * from test_update_column;
 x | y | z | data 
---+---+---+------
 1 | 2 | 4 |  100
 3 | 3 | 6 |  100
(2 rows)

--T19
drop foreign table test_update_column;
create foreign table test_update_column (x int primary key, y int not null, z int not null, data int);
NOTICE:  CREATE FOREIGN TABLE / PRIMARY KEY will create constraint "test_update_column_pkey" for foreign table "test_update_column"
create index idx1 on test_update_column (y);
create unique index idx2 on test_update_column (z);
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,4,100);
begin;
delete from test_update_column;
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,5,100);
update test_update_column set z=z+1;
delete from test_update_column;
insert into test_update_column values (1,2,3,100);
insert into test_update_column values (3,3,5,100);
update test_update_column set z=z+1;
alter foreign table test_update_column drop column y;
commit;
select * from test_update_column;
 x | z | data 
---+---+------
 1 | 4 |  100
 3 | 6 |  100
(2 rows)

drop foreign table test_update_column;
